pkg/reconciler/apiserversource/controller.go

```diff
@@ -59,6 +61,9 @@ func NewController(
 	namespaceInformer := namespace.Get(ctx)
 	serviceaccountInformer := serviceaccountinformer.Get(ctx)
 
+	roleInformer := roleinformer.Get(ctx)
+	rolebindingInformer := rolebindinginformer.Get(ctx)
```

Comment:
Would it be possible to use filtered informers ?

Reply:
Should I change something and do codegen? as I can see for other components, like pkg/client/injection/informers/messaging/v1/subscription/filtered, there is a directory contains filterted informer

@creydr

Reply:
Hey @Leo6Leo,
no need to generate them. They are already available: https://github.com/knative/pkg/tree/main/client/injection/kube/informers/rbac/v1/role/filtered and https://github.com/knative/pkg/tree/main/client/injection/kube/informers/rbac/v1/rolebinding/filtered.

You might only need to rerun `./hack/update-deps.sh` after you imported them to revendor the files.

Reply:
@creydr Tried to implement the filteredInformer, but there is a weird problem and I couldn't figure out what's wrong.

`error:panic: factory: Unable to fetch labelkey from context.`

I used the debugger log, and can confirm that withSelectors function is called, but not sure why the key is still not found.

Reply:
@Leo6Leo: I think you need to import `filtered "knative.dev/pkg/client/injection/kube/informers/factory/filtered"` instead of `filtered "knative.dev/eventing/pkg/client/injection/informers/factory/filtered"`, as filtered role&rolebindinginformer use the `LabelKey{}` from there.

@pierDipi can you give some hints on the filtered informers? I am not sure on how to register the selector. IIUC, we need to register the selector, before the informers get setup. Anyhow this would mean, we would need to register it in the controller main.go and IIUC this selector would be used for all filtered informers. So I think I am missing here something.

Reply:
That's correct, however, as I see from the code, each selector is used for a separate "watch"

```
	labelSelectors := untyped.([]string)
	for _, selector := range labelSelectors {
		selectorVal := selector
		opts := []externalversions.SharedInformerOption{}
		if injection.HasNamespaceScope(ctx) {
			opts = append(opts, externalversions.WithNamespace(injection.GetNamespaceScope(ctx)))
		}
		opts = append(opts, externalversions.WithTweakListOptions(func(l *v1.ListOptions) {
			l.LabelSelector = selectorVal
		}))
		ctx = context.WithValue(ctx, Key{Selector: selectorVal},
			externalversions.NewSharedInformerFactoryWithOptions(c, controller.GetResyncPeriod(ctx), opts...))
	}
```

so yes, in the controller main we would need to register selectors for each individual controllers

Reply:
We should try to see if it works for us

Reply:
Same for individual informers

```
	labelSelectors := untyped.([]string)
	infs := []controller.Informer{}
	for _, selector := range labelSelectors {
		f := factoryfiltered.Get(ctx, selector)
		inf := f.Eventing().V1beta3().EventTypes()
		ctx = context.WithValue(ctx, filtered.Key{Selector: selector}, inf)
		infs = append(infs, inf.Informer())
	}
```

Reply:
Ok, then I think I understood correctly :) The thing that I see, is that we can't specify for which filtered informer we want to register the selector.
e.g. when we register a selector for the oidc-serviceaccount and one for the api-server-source role/rolebinding (e.g. `filteredinformerfactory.WithSelectors(ctx, "oidc-serviceaccount", "role=oidc-token-creator")`) and later import the filtered informers for SA, role and rolebinding it will create two filtered informers for each of them.